applications/users/tests/test_user_update.py

```diff
@@ -0,0 +1,98 @@
+import factory
+import io
+from faker import Faker
+from PIL import Image
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+from rest_framework.views import status
+
+from .factories import BaseUserFactory, UserFactory
+
+
+class UserFieldsUpdateTests(APITestCase):
+    fake = Faker()
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.url = reverse('rest_user_details')
+
+    def setUp(self):
+        self.email = self.fake.email()
+        self.user = UserFactory(email=self.email, confirmed=True)
+        self.params = factory.build(dict, FACTORY_CLASS=BaseUserFactory, email=self.email)
+
+    def call_update(self):
+        self.client.force_authenticate(user=self.user)
+        return self.client.put(self.url, self.params)
+
+    def test_all_params_valid_respond_success(self):
+        response = self.call_update()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            [key for key in response.json()],
+            ['id', 'email', 'name', 'gender', 'picture', ]
+        )
+
+    def test_all_params_valid_save_data(self):
+        response = self.call_update()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.params['email'], self.user.email)
+        self.assertEqual(self.params['name'], self.user.name)
+        self.assertEqual(self.params['gender'], self.user.gender)
```

Comment:
Yo should test that the user has the changed values in the database

Reply:
As Bruno said, you should get the user from the db and check against that data.